core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/DefaultCodecRegistry.java

```diff
@@ -159,7 +159,13 @@ public boolean equals(Object other) {
 
     @Override
     public int hashCode() {
-      return Objects.hash(cqlType, javaType, isJavaCovariant);
+      // NOTE: inlined Objects.hash for performance reasons (avoid Object[] allocation
+      // seen in profiler allocation traces)
+      int h1 = Objects.hashCode(cqlType);
+      int h2 = Objects.hashCode(javaType);
+      int h3 = Boolean.hashCode(isJavaCovariant);
+
+      return 31 * (31 * h1 + h2) + h3;
```

Comment:
According to [List.hashCode()](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#hashCode--)
```java
     int hashCode = 1;
     for (E e : list)
         hashCode = 31*hashCode + (e==null ? 0 : e.hashCode());
```
Isn't it `((31+h1)*31+h2)*31+h3`?

Reply:
Changed as suggested.